### PR TITLE
WIP: update to new @microsoft/vscode-azext-utils

### PR DESCRIPTION
**Pull request by StephenWeatherford**
This is the first and hopefully biggest step to updating to newest dependencies, and brings us fully up to date with the latest telemetry package.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
     "homepage": "https://github.com/Microsoft/vscode-azurearmtools/blob/main/README.md",
     "license": "SEE LICENSE IN LICENSE.md",
     "icon": "AzureRMTools128x128.png",
-    "aiKey": "AIF-d9b70cd4-b9f9-4d70-929b-a071c400b217",
+    "aiKey": "0c6ae279ed8443289764825290e4f9e2-1a736e7c-1324-4338-be46-fc2a58ae4d14-7255",
```
Comment (StephenWeatherford) [inline package.json:43]: new telemetry assembly no longer supports AIF keys

File: src/extension.ts [hunk 2]
```diff
@@ -95,7 +95,6 @@ export async function activateInternal(context: vscode.ExtensionContext, perfSta
         ext.extensionStartupComplete = false;
 
         ext.context = context;
-        ext.ui = new AzureUserInput(context.globalState);
```
Comment (StephenWeatherford) [inline src/extension.ts:98]: see above comment

File: src/util/deepClone.ts [hunk 3]
```diff
@@ -25,7 +25,7 @@ export function deepClone<T extends {}>(value: T): T {
         // tslint:disable-next-line:no-for-in // Grandfathered in
         for (let propertyName in value) {
             // tslint:disable-next-line: no-unsafe-any no-any
-            if (value.hasOwnProperty(propertyName)) {
+            if (Object.prototype.hasOwnProperty.call(value, propertyName)) {
```
Comment (StephenWeatherford) [inline src/util/deepClone.ts:28]: side effect of newer typescript

File: package.json [hunk 4]
```diff
@@ -637,21 +637,25 @@
...
         "@types/rimraf": "2.0.2",
-        "@types/shelljs": "0.8.12",
+        "@types/shelljs": "0.8.15",
         "@types/vscode": "^1.62.0",
-        "@types/webpack": "^4.41.26",
+        "@types/vinyl": "^2.0.12",
+        "@types/webpack": "^5.28.5",
         "@vscode/test-cli": "0.0.10",
         "@vscode/test-electron": "^2.4.1",
+        "@microsoft/vscode-azext-dev": "^2.0.6",
```
Comment (StephenWeatherford) [inline package.json:653]: This replaces vscode-azureextensiondev

File: package.json [hunk 5]
```diff
@@ -661,17 +665,16 @@
...
+        "typescript": "^5.0.3",
         "vsce": "^1.87.0",
-        "vscode-azureextensiondev": "^0.5.0",
-        "webpack": "^4.46.0",
-        "webpack-cli": "^3.3.12"
+        "webpack": "^5.74.0",
+        "webpack-cli": "^4.6.0"
     },
     "dependencies": {
+        "@microsoft/vscode-azext-utils": "^2.0.0",
```
Comment (StephenWeatherford) [inline package.json:674]: This replaces vscode-azureextensionui.

Getting the remaining dependencies to be compatible with these new ones and each other was not easy.  Next I'll update @microsoft/vscode-azext-utils to its latest.

**Review by AbhitejJohn — APPROVED** [review]

**Pull request merged**